### PR TITLE
Migrate GO to Vite

### PR DESCRIPTION
**Pull request by nguyentvan7**
## Describe your changes

Migrate `frontend` from webpack to Vite for improved dev experience.
- [x] Migrate translation/assets
- [x] Migrate tests
- [x] Migrate notification.mp3 and any other misc assets
- [x] Make sure scanner is using trained model

## Issue or discord link

- 

## Testing/validation

* For scanner, I changed the model name without changing the config, and noted the scanner threw a 404 error. I changed it back to match, and the scanner no longer threw a 404, so this validates the file is found and the proper model is being used
* Verified that translations appeared as expected
* Verified that worker-related flows were working
* Verified that notification sound will play
* Verified that hot module reloading was working as expected on changes in the dev server

## Checklist before requesting a review (leave this PR as draft if any part of this list is not done.)

- [ ] I have commented my code in hard-to understand areas.
- [ ] I have made corresponding changes to README or wiki.
- [ ] For front-end changes, I have updated the corresponding English translations.
- [ ] I have run `yarn run mini-ci` locally to validate format and lint.
- [ ] If I have added a new library or app, I have updated the deployment scripts to ignore changes as needed

### CODE REVIEW

File: apps/frontend/src/app/App.scss [hunk 1]
```diff
@@ -51,8 +53,8 @@ body {
     $random-x: random(1000000) * 0.0001vw;
     $random-offset: random_range(-100000, 100000) * 0.0001vw;
     $random-x-end: $random-x + $random-offset;
-    $random-x-end-yoyo: $random-x + ($random-offset / 2);
```
Comment (frzyc) [inline apps/frontend/src/app/App.scss:54]: can you explain why this was done?

File: package.json [hunk 2]
```diff
@@ -82,6 +82,7 @@
     "vite": "^4.3.9",
     "vite-plugin-dts": "2.3.0",
     "vite-plugin-eslint": "^1.8.1",
+    "vite-plugin-static-copy": "^1.0.1",
```
Comment (frzyc) [inline package.json:85]: Can you explain why we can't use `public` directory? https://vitejs.dev/guide/assets.html#the-public-directory

File: libs/gi/db/src/Database/DataEntries/DBMetaEntry.ts [hunk 3]
```diff
@@ -2,11 +2,11 @@ import type { Database } from '@genshin-optimizer/common/database'
 import type { GenderKey } from '@genshin-optimizer/gi/consts'
 import { allGenderKeys } from '@genshin-optimizer/gi/consts'
 import type { IGOOD } from '@genshin-optimizer/gi/good'
-import { DataEntry } from '../DataEntry'
 import type { ArtCharDatabase } from '../ArtCharDatabase'
+import { DataEntry } from '../DataEntry'
 import type { IGO, ImportResult } from '../exim'
 
-interface IDBMeta {
+export interface IDBMeta {
```
Comment (frzyc) [inline libs/gi/db/src/Database/DataEntries/DBMetaEntry.ts:9]: Why are we exporting this interface? its only used locally in this file?

File: libs/gi/db/src/Database/DataManagers/CharMetaDataManager.ts [hunk 4]
```diff
@@ -5,10 +5,10 @@ import type {
   SubstatKey,
 } from '@genshin-optimizer/gi/consts'
 import { allSubstatKeys, allTravelerKeys } from '@genshin-optimizer/gi/consts'
-import { DataManager } from '../DataManager'
 import type { ArtCharDatabase } from '../ArtCharDatabase'
+import { DataManager } from '../DataManager'
 
-interface ICharMeta {
+export interface ICharMeta {
```
Comment (frzyc) [inline libs/gi/db/src/Database/DataManagers/CharMetaDataManager.ts:11]: Why are we exporting this interface? its only used locally in this file?

File: apps/frontend/vite.config.mts [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+  server: {
+    port: 4200,
+    host: 'localhost',
+    // Allows workers to fetch the needed .ts file
+    fs: {
+      allow: ['../../'],
+    },
+  },
+
+  preview: {
```
Comment (frzyc) [inline apps/frontend/vite.config.mts:24]: What is this `preview` used for?

File: apps/frontend/vite.config.mts [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+
+  plugins: [
+    viteTsConfigPaths({
+      root: '../../',
+    }),
+    // Enables proper hot module reloading
+    react({
+      include: '**/*.tsx',
+    }),
+    // Nx executor for vite does not support `assets` prop for copying files.
```
Comment (frzyc) [inline apps/frontend/vite.config.mts:37]: can you link to an nx issue here so we can track this support in the future?

**Review by frzyc — COMMENTED** [review]
Some questions/comments.
Can you also include some source graph comparison and whether if the payload size is significantly increased?

File: package.json (hunk 2 above)
Comment (nguyentvan7) [reply package.json:85]: I tried using the public directory, but the assets weren't loading for some reason. However, if I copy it outside the public directory, into our own `assets` directory, it works.

Also this copy plugin is to do what nx does with the `assets` config in `project.json` for the old webpack version, i.e. copy the assets from one spot to another when building/serving. But nx doesn't support the assets config for vite build for whatever reason

File: apps/frontend/vite.config.mts (hunk 5 above)
Comment (nguyentvan7) [reply apps/frontend/vite.config.mts:24]: https://vitejs.dev/guide/static-deploy.html#testing-the-app-locally

Preview uses the build in the `dist` folder and serves that locally. so you can test prod build locally

File: apps/frontend/vite.config.mts (hunk 6 above)
Comment (nguyentvan7) [reply apps/frontend/vite.config.mts:37]: There is none, not sure if it is an oversight or on purpose. I can create one

File: apps/frontend/src/app/App.scss (hunk 1 above)
Comment (nguyentvan7) [reply apps/frontend/src/app/App.scss:54]: This was throwing an error for some reason, division isnt supported in sass like this

File: package.json (hunk 2 above)
Comment (frzyc) [reply package.json:85]: Can you put something in the README about how to configure and add new assets in the README?

**Comment by nguyentvan7** [comment]
Webpack `dist/apps/frontend` size: 181M
Vite `dist/apps/frontend` size: 177M

Webpack graph: 
image


Vite graph
image

**Review by frzyc — APPROVED** [review]
lets go

**Pull request merged**